projects/step-core/src/lib/services/view-registry.service.ts

```diff
@@ -146,15 +153,99 @@ export class ViewRegistryService {
     this.registerViewWithConfig(viewId, template, { isPublicView: isPublicView });
   }
 
-  registerRoute(route: Route): void {
-    const root = this._router.config.find((parent) => parent.path === 'root');
+  getChildrenRouteInfo(parentPath: string): SubRouteData[] {
+    const parentChildren = this.getRouteParentChildren(parentPath);
+    return parentChildren
+      .map((child) => {
+        const data = child.data?.[SUB_ROUTE_DATA];
+        if (!data) {
+          return undefined;
+        }
+        const path = child.path;
+        return { ...data, path } as SubRouteData;
+      })
+      .filter((data) => !!data)
+      .sort((a, b) => {
+        const weightA = a?.weight ?? 1;
+        const weightB = b?.weight ?? 1;
+        return weightA - weightB;
+      }) as SubRouteData[];
+  }
+
+  private getRouteParentChildren(parentPath: string): Routes {
+    const root = this._router.config.find((route) => route.path === 'root')!;
+    let parentChildren = root.children!;
+
+    if (!parentPath) {
+      return parentChildren;
+    }
+
+    const parts = parentPath.split('/');
+
+    for (const parentPathPart of parts) {
+      const parent = parentChildren.find((route) => route.path === parentPathPart);
+
+      if (parent) {
+        parent.children = parent.children ?? [];
+        parentChildren = parent.children;
+        continue;
+      }
+
+      if (this.temporaryRouteChildren.has(parentPathPart)) {
+        parentChildren = this.temporaryRouteChildren.get(parentPathPart)!;
+      } else {
+        parentChildren = [];
+        this.temporaryRouteChildren.set(parentPathPart, parentChildren);
+      }
+    }
+
+    return parentChildren;
+  }
+
+  registerRoute(route: Route, { parentPath, label, weight }: SubRouterConfig = {}): void {
+    const root = this._router.config.find((route) => route.path === 'root');
```

Comment:
I would prefer to move all parameters into one Object, probably the best way would be by extending Route to include parentPath, label, weight

Reply:
@neogucky 
I don't think, that it would be correct to unite in one object.
`route` object can contain the whole subtrees, adding `parentPath` field to it, would bring some kind of duality in route declaration - that we can define both parent and children.
In our case `parentPath` is required only for those cases, when plugin extends a view.
About `label` and `weight`, if you inspect the code further you will see, that these fields are merged in route's data object.
So in other words this `SubRouteConfig` is just a syntax sugar to simplify the route's registration